templates/gateway.tmpl.yaml

```diff
@@ -0,0 +1,139 @@
+  server.key: LS0tLS1CRUdJTiBFQyBQUklWQVRFIEtFWS0tLS0tCk1IY0NBUUVFSVBVa2w0NW0xeWxCT0xMazBITW85WVFwZ3cyRGQxYTFUWFJYYnI3cWRuOXlvQW9HQ0NxR1NNNDkKQXdFSG9VUURRZ0FFdTdCRmpiM3ZDakxXTzF4dnhhYTlkQWxKck52Y1RSYkN2TXJYVDlxK2tqcyt6MUZ3cHJPNQpjMGpkK1JRQW9kUnNQbGZ5Rnc1N3FQMmpPem5obm9FeUlnPT0KLS0tLS1FTkQgRUMgUFJJVkFURSBLRVktLS0tLQo=
+kind: Secret
+metadata:
+  creationTimestamp: 2017-12-21T07:39:01Z
+  name: gateway-tls
+  namespace: default
+  selfLink: /api/v1/namespaces/default/secrets/gateway-tls
+type: Opaque
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: gateway-deployment
+spec:
+  selector:
+    matchLabels:
+      app: gateway
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: gateway
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.13.7
+        command:
+        - nginx-debug
+        - -g
+        - 'daemon off;'
+        volumeMounts:
+        - name: nginx-config
+          mountPath: /etc/nginx/conf.d/
+        - name: gateway-tls
+          mountPath: /etc/ssl/gateway/
+          readOnly: true
+        ports:
+        - containerPort: 80
+        - containerPort: 8443
+      volumes:
+      - name: nginx-config
+        configMap:
+          name: gateway-config
+      - name: gateway-tls
+        secret:
+          secretName: gateway-tls
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: gateway-service
+spec:
+  ports:
+     - name: http
+       port: 80
+       protocol: TCP
+       targetPort: 80
+     - name: https
+       port: 8443
+       protocol: TCP
+       targetPort: 8443
+       nodePort: 30443
```

Comment:
This port is in the k8s default reserved port range `30000-32767`. I would pick a port outside that range.

Reply:
Exactly, node port is required to be in this range